### PR TITLE
Controllers testing 

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of blog app we have added the following requirements :

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers_spec/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+require 'factory_bot_rails'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (Ismailco) [inline spec/controllers_spec/posts_controller_spec.rb:4]: - [ ] Kindly make sure to implement the tests for the `Posts` show action also

File: spec/controllers_spec/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  before(:each) do
+    # Clean up the test database before each test
```
Comment (Ismailco) [inline spec/controllers_spec/posts_controller_spec.rb]: - [ ] Kindly consider removing all the comments that are not needed in production from your code base, these comments are useful in the development process but they should be removed when you push your code to GitHub

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @ridabensalem, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @ridabensalem,

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**